# Library Imports

In [1]:
from gameinsights import Collector

In [2]:
import pandas as pd

In [3]:
# import secrets
from dotenv import load_dotenv
import os
load_dotenv()

True

# Collecting data using GameInsights

In [4]:
steam_api_key = os.getenv("STEAM_API_KEY")

In [5]:
collector = Collector(steam_api_key=steam_api_key)

In [6]:
eiyuden_appid = "1658280"

## Collecting review data

In [7]:
reviews = collector.get_game_review(steam_appid=eiyuden_appid, verbose=False, review_only=True)

In [13]:
reviews.shape

(4100, 21)

In [9]:
reviews.columns

Index(['recommendation_id', 'author_steamid', 'author_num_games_owned',
       'author_num_reviews', 'author_playtime_forever',
       'author_playtime_last_two_weeks', 'author_playtime_at_review',
       'author_last_played', 'language', 'review', 'timestamp_created',
       'timestamp_updated', 'voted_up', 'votes_up', 'votes_funny',
       'weighted_vote_score', 'comment_count', 'steam_purchase',
       'received_for_free', 'written_during_early_access',
       'primarily_steam_deck'],
      dtype='object')

In [37]:
# save review to csv
reviews.to_csv("data/eiyuden_reviews.csv", index=False)

In [24]:
# save reviews to parquet
reviews.to_parquet("data/eiyuden_reviews.parquet", index=False)

## Collecting user data

In [23]:
reviews = pd.read_csv("data/eiyuden_reviews.csv")

In [8]:
reviews.head()

recommendation_id     author_steamid  author_num_games_owned  \
0          201952284  76561198045255045                       0   
1          201940210  76561198113965734                       0   
2          201914252  76561198138897509                       0   
3          201894653  76561198368799839                    6496   
4          201825659  76561198449223262                       0   

   author_num_reviews  author_playtime_forever  \
0                  10                        9   
1                   2                     2734   
2                   7                     1530   
3                 540                     8511   
4                   2                     5018   

   author_playtime_last_two_weeks  author_playtime_at_review  \
0                               9                          9   
1                            1447                       2734   
2                             611                       1377   
3                             366                       8511   
4                            1605                       5018   

   author_last_played  language  \
0          1754967253   english   
1          1754703861   english   
2          1754932590   italian   
3          1754893622   koreana   
4          1754809816  japanese   

                                              review  ...  timestamp_updated  \
0  By the 4th text box in the game I noticed some...  ...         1754968462   
1  Needs a x2 , x3  speed key. Would shave off 70...  ...         1754950716   
2           per ora bella storia e grafica gradevole  ...         1754923391   
3  - 이 게임의 가장 큰 문제점은 환상수호전2라는 벽이 너무 거대하다는 거다.\n본인...  ...         1754896546   
4  仲間全員集めてレースやカードゲーム等のミニゲームにも手を付けてプレイ時間約８０時間でクリアし...  ...         1754812678   

   voted_up  votes_up  votes_funny  weighted_vote_score  comment_count  \
0     False         0            0             0.500000              0   
1     False         0            0             0.500000              0   
2      True         0            0             0.500000              0   
3      True         8            1             0.619325              0   
4      True         0            0             0.500000              0   

   steam_purchase  received_for_free  written_during_early_access  \
0           False              False                        False   
1            True              False                        False   
2            True              False                        False   
3           False              False                        False   
4           False              False                        False   

   primarily_steam_deck  
0                 False  
1                 False  
2                 False  
3                 False  
4                 False  

[5 rows x 21 columns]

In [9]:
# extract author_steamid to collect user data
authors = reviews['author_steamid'].to_list()

In [10]:
len(authors)

4100

In [ ]:
user_data = collector.get_user_data(
    steamids=authors,
    include_free_games=True,
    verbose=True,
)

In [12]:
user_data

steamid  community_visibility_state  profile_state  \
0     76561198045255045                           2            1.0   
1     76561198113965734                           1            1.0   
2     76561198138897509                           3            1.0   
3     76561198368799839                           3            1.0   
4     76561198449223262                           3            1.0   
...                 ...                         ...            ...   
4095  76561198974329255                           3            1.0   
4096  76561198243956334                           3            1.0   
4097  76561198089697358                           3            1.0   
4098  76561198035388233                           3            1.0   
4099  76561198055295470                           3            1.0   

      persona_name                                        profile_url  \
0          Dojilol             https://steamcommunity.com/id/Dojilol/   
1              123  https://steamcommunity.com/profiles/7656119811...   
2           Pino70  https://steamcommunity.com/profiles/7656119813...   
3            oci51  https://steamcommunity.com/profiles/7656119836...   
4      Rock in JPN  https://steamcommunity.com/profiles/7656119844...   
...            ...                                                ...   
4095       CR4V3N1  https://steamcommunity.com/profiles/7656119897...   
4096  Onion Knight  https://steamcommunity.com/profiles/7656119824...   
4097   Crystalizen           https://steamcommunity.com/id/crystlfst/   
4098        ubri04              https://steamcommunity.com/id/ubri04/   
4099     Althuriaz  https://steamcommunity.com/profiles/7656119805...   

      last_log_off real_name  time_created loc_country_code loc_state_code  \
0              NaN      None           NaN             None           None   
1              NaN      None           NaN             None           None   
2              NaN  Giuseppe  1.401729e+09               IT             05   
3              NaN       oci  1.487895e+09               KR           None   
4              NaN      None  1.511958e+09               JP           None   
...            ...       ...           ...              ...            ...   
4095           NaN      None  1.561924e+09               US             TX   
4096           NaN     tommy  1.439213e+09             None           None   
4097           NaN      None  1.366967e+09               KR           None   
4098           NaN    Aubrey  1.293036e+09             None           None   
4099           NaN      None  1.324750e+09             None           None   

      loc_city_id                                        owned_games  \
0             NaN                                                 {}   
1             NaN                                                 {}   
2         24453.0                     {'game_count': 0, 'games': []}   
3             NaN  {'game_count': 6658, 'games': [{'appid': 1610,...   
4             NaN                     {'game_count': 0, 'games': []}   
...           ...                                                ...   
4095       3577.0                     {'game_count': 0, 'games': []}   
4096          NaN                     {'game_count': 0, 'games': []}   
4097          NaN                     {'game_count': 0, 'games': []}   
4098          NaN  {'game_count': 495, 'games': [{'appid': 3330, ...   
4099          NaN                     {'game_count': 0, 'games': []}   

                                  recently_played_games  
0                                                    {}  
1                                                    {}  
2     {'games_count': 0, 'total_playtime_2weeks': 0,...  
3     {'games_count': 9, 'total_playtime_2weeks': 73...  
4     {'games_count': 0, 'total_playtime_2weeks': 0,...  
...                                                 ...  
4095  {'games_count': 0, 'total_playtime_2weeks': 0,...  
4096  {'games_count': 0, 'total_

In [13]:
# save data to csv
user_data.to_csv("data/eiyuden_user_data.csv", index=False)

In [25]:
# sasve data to parquet
user_data.to_parquet("data/eiyuden_user_data.parquet", index=False)

## Collecting game data

In [7]:
# import user data parquet
user_data = pd.read_parquet("data/eiyuden_user_data.parquet")

In [8]:
user_data.head()

steamid  community_visibility_state  profile_state persona_name  \
0  76561198045255045                           2            1.0      Dojilol   
1  76561198113965734                           1            1.0          123   
2  76561198138897509                           3            1.0       Pino70   
3  76561198368799839                           3            1.0        oci51   
4  76561198449223262                           3            1.0  Rock in JPN   

                                         profile_url  last_log_off real_name  \
0             https://steamcommunity.com/id/Dojilol/           NaN      None   
1  https://steamcommunity.com/profiles/7656119811...           NaN      None   
2  https://steamcommunity.com/profiles/7656119813...           NaN  Giuseppe   
3  https://steamcommunity.com/profiles/7656119836...           NaN       oci   
4  https://steamcommunity.com/profiles/7656119844...           NaN      None   

   time_created loc_country_code loc_state_code  loc_city_id  \
0           NaN             None           None          NaN   
1           NaN             None           None          NaN   
2  1.401729e+09               IT             05      24453.0   
3  1.487895e+09               KR           None          NaN   
4  1.511958e+09               JP           None          NaN   

                                         owned_games  \
0                {'game_count': None, 'games': None}   
1                {'game_count': None, 'games': None}   
2                   {'game_count': 0.0, 'games': []}   
3  {'game_count': 6658.0, 'games': [{'appid': 161...   
4                   {'game_count': 0.0, 'games': []}   

                               recently_played_games  
0  {'games': None, 'games_count': None, 'total_pl...  
1  {'games': None, 'games_count': None, 'total_pl...  
2  {'games': [], 'games_count': 0.0, 'total_playt...  
3  {'games': [{'appid': 978300, 'name': 'Saints R...  
4  {'games': [], 'games_count': 0.0, 'total_playt...

In [9]:
user_data.columns

Index(['steamid', 'community_visibility_state', 'profile_state',
       'persona_name', 'profile_url', 'last_log_off', 'real_name',
       'time_created', 'loc_country_code', 'loc_state_code', 'loc_city_id',
       'owned_games', 'recently_played_games'],
      dtype='object')

In [10]:
user_data.dtypes

steamid                         int64
community_visibility_state      int64
profile_state                 float64
persona_name                   object
profile_url                    object
last_log_off                  float64
real_name                      object
time_created                  float64
loc_country_code               object
loc_state_code                 object
loc_city_id                   float64
owned_games                    object
recently_played_games          object
dtype: object

In [11]:
user_data.shape

(4100, 13)

In [12]:
# let's filter users with community_visibility_state == 3 (public/visible to everyone)
public_user = user_data[user_data['community_visibility_state'] == 3]

In [13]:
public_user.shape

(3222, 13)

In [14]:
# there shouldn't be users with owned_games' game_count == 0 since the list of users who reviewed eiyuden chronicle
# therefore, they should at least own a game (whether they received it for free or not)
# but somehow some of the users listed here has games_count == 0
# but maybe they refunded the game once they reviewed it? so let's filter them off
public_user_has_games = public_user[
    public_user['owned_games'].apply(lambda x: x.get('game_count', 0) > 0)
]

In [15]:
public_user_has_games.shape

(1495, 13)

In [16]:
# from 3222 rows down to 1495, that's surprisingly A LOT lol but let's see
public_user_has_games.head()

steamid  community_visibility_state  profile_state persona_name  \
3  76561198368799839                           3            1.0        oci51   
5  76561198971574247                           3            1.0           依然   
7  76561198123424467                           3            1.0       108Hvs   
8  76561198102397621                           3            1.0        Malam   
9  76561198869364047                           3            1.0        ScyRo   

                                         profile_url  last_log_off  \
3  https://steamcommunity.com/profiles/7656119836...           NaN   
5  https://steamcommunity.com/profiles/7656119897...           NaN   
7              https://steamcommunity.com/id/108Hvs/           NaN   
8  https://steamcommunity.com/profiles/7656119810...           NaN   
9  https://steamcommunity.com/profiles/7656119886...           NaN   

         real_name  time_created loc_country_code loc_state_code  loc_city_id  \
3              oci  1.487895e+09               KR           None          NaN   
5             None  1.561081e+09               CN           None          NaN   
7  Haffipul Saddad  1.390055e+09               ID             30          NaN   
8             None  1.376225e+09               KR           None          NaN   
9    Simon Rosales  1.541499e+09               US           None          NaN   

                                         owned_games  \
3  {'game_count': 6658.0, 'games': [{'appid': 161...   
5  {'game_count': 636.0, 'games': [{'appid': 10, ...   
7  {'game_count': 128.0, 'games': [{'appid': 4000...   
8  {'game_count': 79.0, 'games': [{'appid': 8870,...   
9  {'game_count': 130.0, 'games': [{'appid': 2090...   

                               recently_played_games  
3  {'games': [{'appid': 978300, 'name': 'Saints R...  
5  {'games': [{'appid': 2277560, 'name': 'WUCHANG...  
7  {'games': [{'appid': 477160, 'name': 'Human Fa...  
8  {'games': [{'appid': 1658280, 'name': 'Eiyuden...  
9  {'games': [{'appid': 1658280, 'name': 'Eiyuden...

In [17]:
# now we need to store ALL the games that user have played to compare with eiyuden chronicle in a set
# only the games that have playtime > 0
games_set = {
    game['appid']
    for owned_games in public_user_has_games['owned_games']
    for game in owned_games.get('games', []) # they all should have 'games' though but just to be safe
    if game.get('playtime_forever', 0) > 0 # the 'playtime_forever' key also should be there but again, just to be safe
}

In [18]:
games_set_list = list(games_set)

In [19]:
len(games_set_list)

30471

30,471 games information needs to be collected, that's A LOT of games but since we can only filter the games based on its tags AFTER we collected the data, we need to collect all the games data first.

In [20]:
game_data = []
counter = 0
for game in games_set_list:
    print(f"fetching idx {counter}, data for appid {game}")
    try:
        data = collector.get_games_data(steam_appids=game, verbose=False)
        game_data.append(data[0])
    except Exception as e:
        print(f"Error fetching data for game {game}: {e}")
    counter += 1

fetching idx 0, data for appid 393220
fetching idx 1, data for appid 262150
fetching idx 2, data for appid 2752520
fetching idx 3, data for appid 10
fetching idx 4, data for appid 655370
fetching idx 5, data for appid 393230
fetching idx 6, data for appid 20
fetching idx 7, data for appid 655380
fetching idx 8, data for appid 393240
fetching idx 9, data for appid 1048600
fetching idx 10, data for appid 3014680
fetching idx 11, data for appid 2097180
fetching idx 12, data for appid 30
fetching idx 13, data for appid 1179680
fetching idx 14, data for appid 40
fetching idx 15, data for appid 262190
fetching idx 16, data for appid 1441840
fetching idx 17, data for appid 50
fetching idx 18, data for appid 1966130
fetching idx 19, data for appid 2228280
fetching idx 20, data for appid 786490
fetching idx 21, data for appid 60
fetching idx 22, data for appid 262210
fetching idx 23, data for appid 70
fetching idx 24, data for appid 2097230
fetching idx 25, data for appid 80
fetching idx 26, da

KeyboardInterrupt: 

In [22]:
game_data_first_cp = pd.DataFrame(game_data)

In [24]:
game_data_first_cp.to_parquet("data/game_data_3340.parquet", index=False)

In [ ]:
# sasve data to parquet
user_data.to_parquet("data/eiyuden_user_data.parquet", index=False)

In [23]:
game_data_first_cp.head()

steam_appid                                        name        developers  \
0      393220                        Super Man Or Monster           [Xform]   
1      262150                           Vanguard Princess  [Tomoaki Sugeno]   
2     2752520  Warrior Maiden Lecia and the Lost Fortress    [Kuroto Zakka]   
3          10                              Counter-Strike           [Valve]   
4      655370                                Train Bandit    [Rusty Moyher]   

       publishers  type price_currency  price_initial  price_final  \
0         [Xform]  game            USD          12.99        12.99   
1     [eigoMANGA]  game            USD           4.99         4.99   
2    [OTAKU Plan]  game            USD          19.99        15.99   
3         [Valve]  game            USD           9.99         9.99   
4  [Wild Rooster]  game            USD           0.99         0.99   

   metacritic_score release_date  ...  count_backlog  count_review  \
0               NaN   2017-10-31  ...           19.0           1.0   
1               NaN   2014-03-03  ...         1425.0          23.0   
2               NaN   2025-05-23  ...            NaN           NaN   
3              88.0   2000-11-01  ...         4881.0         350.0   
4               NaN   2017-10-12  ...            7.0           2.0   

   count_playing  count_retired  languages              platforms  \
0            0.0            3.0         []              [windows]   
1           11.0          200.0         []  [windows, mac, linux]   
2            NaN            NaN         []              [windows]   
3           68.0         2459.0         []  [windows, mac, linux]   
4            0.0            0.0         []         [windows, mac]   

                                          categories  \
0  [Single-player, Multi-player, PvP, Shared/Spli...   
1  [Single-player, Multi-player, PvP, Shared/Spli...   
2  [Single-player, Full controller support, Steam...   
3  [Multi-player, PvP, Online PvP, Shared/Split S...   
4  [Single-player, Steam Achievements, Full contr...   

                            genres  \
0       [Action, Adventure, Indie]   
1          [Action, Casual, Indie]   
2  [Action, Adventure, Indie, RPG]   
3                         [Action]   
4                  [Action, Indie]   

                                                tags  \
0  [Action, Indie, Adventure, Destruction, Voxel,...   
1  [Anime, Fighting, 2D Fighter, Indie, Female Pr...   
2  [Nudity, Sexual Content, Action, NSFW, Adventu...   
3  [Action, FPS, Multiplayer, Shooter, Classic, T...   
4  [Indie, Action, Pixel Graphics, 2D, Retro, Arc...   

                                      content_rating  
0  [{'rating_type': 'dejus', 'rating': 'l'}, {'ra...  
1  [{'rating_type': 'dejus', 'rating': 'l'}, {'ra...  
2  [{'rating_type': 'dejus', 'rating': '16'}, {'r...  
3  [{'rating_type': 'usk', 'rating': '16'}, {'rat...  
4  [{'rating_type': 'dejus', 'rating': '14'}, {'r...  

[5 rows x 51 columns]

In [ ]:
game_data = collector.get_games_data(
    steam_appids=list(games_set),
    verbose=True, # we don't really need to see the progress
)

INFO - SteamStore: Fetching data for appid 393220.
INFO - Gamalytic: Fetching data for appid 393220.
INFO - SteamSpy: Fetching data for appid 393220.
INFO - SteamCharts: Fetching active player data for appid 393220.
INFO - SteamReview: Fetching review data for appid 393220.
INFO - SteamAchievements: Fetch data for appid 393220.
INFO - HowLongToBeat: Fetching data for game 'Super Man Or Monster'
INFO - SteamStore: Fetching data for appid 262150.
INFO - Gamalytic: Fetching data for appid 262150.
INFO - SteamSpy: Fetching data for appid 262150.
INFO - SteamCharts: Fetching active player data for appid 262150.
INFO - SteamReview: Fetching review data for appid 262150.
INFO - SteamAchievements: Fetch data for appid 262150.
INFO - HowLongToBeat: Fetching data for game 'Vanguard Princess'
INFO - SteamStore: Fetching data for appid 2752520.
INFO - Gamalytic: Fetching data for appid 2752520.
INFO - SteamSpy: Fetching data for appid 2752520.
INFO - SteamCharts: Fetching active player data for ap